internal/pkg/deploy/cloudformation/stack/scheduled_job.go

```diff
@@ -0,0 +1,299 @@
+
+	every = "@every "
+)
+
+// NewScheduledJob creates a new ScheduledJob stack from a manifest file.
+func NewScheduledJob(mft *manifest.ScheduledJob, env, app string, rc RuntimeConfig) (*ScheduledJob, error) {
+	parser := template.New()
+	addons, err := addon.New(aws.StringValue(mft.Name))
+	if err != nil {
+		return nil, fmt.Errorf("new addons: %w", err)
+	}
+	envManifest, err := mft.ApplyEnv(env)
+	if err != nil {
+		return nil, fmt.Errorf("apply environment %s override: %w", env, err)
+	}
+	return &ScheduledJob{
+		wkld: &wkld{
+			name:   aws.StringValue(mft.Name),
+			env:    env,
+			app:    app,
+			tc:     envManifest.ScheduledJobConfig.TaskConfig,
+			rc:     rc,
+			parser: parser,
+			addons: addons,
+		},
+		manifest: envManifest,
+
+		parser: parser,
+	}, nil
+}
+
+// Template returns the CloudFormation template for the scheduled job.
+func (j *ScheduledJob) Template() (string, error) {
+
+	outputs, err := j.addonsOutputs()
+	if err != nil {
+		return "", err
+	}
+
+	sidecars, err := j.manifest.Sidecar.Options()
+	if err != nil {
+		return "", fmt.Errorf("convert the sidecar configuration for job %s: %w", j.name, err)
+	}
+
+	schedule, err := j.awsSchedule()
+	if err != nil {
+		return "", fmt.Errorf("convert schedule for job %s: %w", j.name, err)
+	}
+
+	stateMachine, err := j.stateMachine()
+	if err != nil {
+		return "", fmt.Errorf("convert retry/timeout config for job %s: %w", j.name, err)
+	}
+
+	content, err := j.parser.ParseScheduledJob(template.WorkloadOpts{
+		Variables:          j.manifest.Variables,
+		Secrets:            j.manifest.Secrets,
+		NestedStack:        outputs,
+		Sidecars:           sidecars,
+		ScheduleExpression: schedule,
+		StateMachine:       stateMachine,
+		LogConfig:          j.manifest.LogConfigOpts(),
+	})
+	if err != nil {
+		return "", fmt.Errorf("parse scheduled job template: %w", err)
+	}
+	return content.String(), nil
+}
+
+// awsSchedule converts the Schedule string to the format required by Cloudwatch Events
+// https://docs.aws.amazon.com/lambda/latest/dg/services-cloudwatchevents-expressions.html
+// Cron expressions must have an sixth "year" field, and must contain at least one ? (either-or)
+// in either day-of-month or day-of-week.
+// Day-of-week expressions are zero-indexed in Golang but one-indexed in AWS.
+// @every cron definition strings are converted to rates.
+// All others become cron expressions.
+func (j *ScheduledJob) awsSchedule() (string, error) {
+	if j.manifest.Schedule == "" {
+		return "", fmt.Errorf("missing required field schedule in manifest for job %s", j.name)
+	}
+
+	// Try parsing the string as a cron expression to validate it.
+	if _, err := cron.ParseStandard(j.manifest.Schedule); err != nil {
+		return "", fmt.Errorf("schedule %s for job %s is not a valid cron expression or definition string", j.manifest.Schedule, j.name)
+	}
+	var scheduleExpression string
+	var err error
+	switch {
+	case strings.HasPrefix(j.manifest.Schedule, every):
```

Comment:
why do we use hasprefix instead of 
```suggestion
	switch j.manifest.Schedule {
	case every:
```

Reply:
definition strings like the @every case are specified as `@every 1h30m` or `@every 6h`. That means we need to check that there's a valid prefix before we parse the duration string. Especially since immediately after this we check for valid predefined schedules.

Reply:
I was dumb...nvm then